apps/frontend/src/app/components/level-indicator/level-indicator.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { XpSystemsService } from '../../services/xp-systems.service';
+
+@Component({
+  selector: 'm-level-indicator',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './level-indicator.component.html'
+})
+export class LevelIndicatorComponent implements OnInit {
+  protected level: number;
+  protected levelColor: string;
+  protected prestige: number;
+  protected prestigeIcon: string;
+  protected textColor: string;
+  protected imageClass: string;
+
+  @Input()
+  totalLevel: number;
+
+  constructor(private readonly xpService: XpSystemsService) {}
+
+  ngOnInit() {
```

Comment:
`ngOnInit` isn't right here, both because the local user service might not have fetched the latest user data when this component is first created, and because the user data might update at some point. User's can't change their XP just by using the site, but we could have cases where they're playing the game at the time as having the site open, and we'd want that to affect this component.

Again it's Angular so there's several ways to do this, but simplest in this case is to just replace `OnInit`/`ngOnInit` with `OnChanges`/`ngOnChanges`. You only have one input, so if that method is called, the level must have updated.